### PR TITLE
#75 use a connection factory with automatic recovery disabled by default

### PR DESCRIPTION
**Pull request by basert**
#75

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

**Comment by michaelklishin** [comment]
@jhalterman I think this is good to be merged. Should a new (point?) release be produced as well?

**Comment by michaelklishin** [comment]
@jhalterman sorry, any thoughts? If you have absolutely no time for Lyra, can you add me as a project owner so that I can at least merge small PRs?

**Comment by basert** [comment]
Anything new on this?

**Comment by michaelklishin** [comment]
I'm trying to reach out to the maintainer via means other than GitHub to obtain the permissions so that user PRs like this one can be merged quickly.

**Review by michaelklishin — APPROVED** [review]

**Comment by DimaGolomozy** [comment]
@michaelklishin 
what about this code [ConnectionOptions.java#L39-L46](https://github.com/basert/lyra/blob/a838fef9008a1487e512f3f3b74a04a4aa533e8d/src/main/java/net/jodah/lyra/ConnectionOptions.java#L39-L46), The user can passes the `ConnectionFactory` with `automaticRecovery` flag enabled and it will not be checked.

I suggest to put an Assertion when the connection is been created, see [my PR](https://github.com/DimaGolomozy/lyra/blob/b1137afadbbdb04410cf3c319c60977fb303a891/src/main/java/net/jodah/lyra/Connections.java#L112-L113).

File: src/main/java/net/jodah/lyra/ConnectionOptions.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ private ConnectionOptions(ConnectionOptions options) {
     nioParams = options.nioParams;
     useNio = options.useNio;
 
-    factory = new ConnectionFactory();
+    factory = makeConnectionFactory();
     factory.setAutomaticRecoveryEnabled(options.factory.isAutomaticRecoveryEnabled());
```
Comment (DimaGolomozy) [inline src/main/java/net/jodah/lyra/ConnectionOptions.java]: Small, but I guess this setter is no longer needed here

**Comment by austin48** [comment]
Does the main README and it's examples need to be updated to reflect that automatic recovery is disabled by default in 0.5.5?  Also, if I had been using a recovery policy, should I remove that from my config because recovery is now disabled?

**Comment by acogoluegnes** [comment]
> Does the main README and it's examples need to be updated to reflect that automatic recovery is disabled by default in 0.5.5?

Java client's automatic recovery is disabled by default, not Lyra's. There's already a [warning](https://github.com/jhalterman/lyra#automatic-resource-recovery) in the readme to make sure automatic recovery is not enabled for both libraries.

> Also, if I had been using a recovery policy, should I remove that from my config because recovery is now disabled?

Disabling automatic recovery in the underlying Java client should not affect Lyra's recovery policy.

Please note [this project is no longer maintained](https://github.com/jhalterman/lyra#project-status).

**Pull request merged**